main_server/server/c_comdef_server.class.php

```diff
@@ -3077,18 +3077,15 @@ public static function GetUserServiceBodies( $in_user_id = null   ///< The ID of
                 $is_editor = null;
                 if ($service_body instanceof c_comdef_service_body) {
                     $editors = $service_body->GetEditors();
-                    
                     if (is_array($editors) && count($editors)) {
                         if (in_array($in_user_id, $editors)) {
                             $is_editor = 'editor';
                         }
                     }
-                    
-                    if ($service_body->GetPrincipalUserID() == $in_user_id) {
+                    if ($service_body->IsUserInServiceBodyHierarchy(self::GetCurrentUserObj())) {
```

Comment:
I could be wrong, but I think `IsUserInServiceBodyHierarchy` will return true if the user is not an owner, but only an editor of a parent service body. I don't think that is what we want. Only "owner" permissions should recurse downwards through the tree, not "editor" permissions.

Reply:
Right, `IsUserInServiceBodyHierarchy` will also return true if the user is not an owner, but only an editor of a parent service body.   But permission to edit meetings does recurse downward through the tree, so for consistency it seems reasonable to do it here also.  For example, if I create a new user `zonewide` and add it to the Full Meeting List Editors for the zone, then `zonewide` can edit a meeting whose service body is an area in the zone.  Seems plausible for `zonewide` to be able to import NAWS world IDs also ....